src/pages/komodo-defi-framework/api/v20/enable_eth_with_tokens/index.mdx

```diff
@@ -14,6 +14,8 @@ Additionally, it supports ERC20 tokens on the ETH chain and associated ERC20 lik
 | mm2                         | integer          | Required if not set in `coins` file. Informs the Komodo DeFi Framework API whether or not the coin is expected to function. Accepted values are `0` or `1`                                                                                                                                                                                   |
 | swap\_contract\_address     | string           | Address of etomic swap smart contract                                                                                                                                                                                                                                                                                                        |
 | fallback\_swap\_contract    | string           | Address of backup etomic swap smart contract                                                                                                                                                                                                                                                                                                 |
+| maker\_swap\_v2\_contract   | string           | Backup address for the maker's swap smart contract                                                                                                                                                                                                                                                                                              |
+| taker\_swap\_v2\_contract   | string           | Backup address for the taker's swap smart contract                                                                                                                                                                                                                                                                 |
```

Comment:
swap_v2 addresses are not backup, right now we have legacy swap protocol and implementing the new one, calling it trading protocol upgrade https://github.com/KomodoPlatform/komodo-defi-framework/issues/1895
`swap_contract_address` field is a legacy adress of etomic swap smart contract.

v2 swap contracts are related to etomic swap contracts implemented for trading protocol upgrade v2
https://github.com/KomodoPlatform/etomic-swap/tree/5e15641cbf41766cd5b37b4d71842c270773f788/contracts

`maker_swap_v2_contract`, `taker_swap_v2_contract` and `nft_maker_swap_v2_contract` addresses are
fields of deployed `EtomicSwapMakerV2`, `EtomicSwapTakerV2` and `EtomicSwapMakerNftV2` contracts
https://github.com/KomodoPlatform/etomic-swap/tree/5e15641cbf41766cd5b37b4d71842c270773f788/contracts

Optional "swap_v2_contracts" field was added in `enable_eth_with_tokens` RPC
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129#discussion_r1661916230
in rust code it looks like this
```rust
pub swap_v2_contracts: Option<SwapV2Contracts>

pub struct SwapV2Contracts {
    pub maker_swap_v2_contract: Address,
    pub taker_swap_v2_contract: Address,
    pub nft_maker_swap_v2_contract: Address,
}
```

if `"use_trading_proto_v2"`is true in mm config (json config which you provide when start program) then `swap_v2_contracts` must be provided

so update the request fields.

Reply:
also you should provide info about optional `"nft_req"` field in activation request

in rust code this field looks like this, it can help you with object names and description in documentation
```rust
nft_req: Option<NftActivationRequest>,

/// Encapsulates the request parameters for NFT activation, specifying the provider to be used.
#[derive(Clone, Deserialize)]
pub struct NftActivationRequest {
    pub provider: NftProviderEnum,
}

/// Defines available NFT providers and their configuration.
#[derive(Clone, Deserialize)]
#[serde(tag = "type", content = "info")]
pub enum NftProviderEnum {
    Moralis {
        url: Url,
        #[serde(default)]
        proxy_auth: bool,
    },
}
```

so in json  request the rust code looks like
```json
    "nft_req": {
      "provider":{
        "type": "Moralis",
        "info": {
          "url":"http://localhost:6150/nft-test",
          "proxy_auth":true
        }
      }
    }
```